src/MongoDB.Analyzer/Core/TypesProcessor.cs

```diff
@@ -73,9 +64,9 @@ public string ProcessTypeSymbol(ITypeSymbol typeSymbol)
         }
 
         var fullTypeName = GetFullName(typeSymbol);
-        if (s_knownBsonTypes.TryGetValue(fullTypeName, out var knowTypeName))
+        if (typeSymbol.IsSupportedBsonType() || typeSymbol.IsSupportedBsonSerializationOption())
```

Comment:
These methods perform `ToDisplayString` twice, while we already have `fullTypeName`.
Lets minimize the lookups and imbed `IsSupportedBsonSerializationOption` into `IsSupportedBsonType` (`startsWith(NamespaceMongoDBBson)`), and receive optional parameter `fullTypeName`.

Reply:
Done